src/Framework/PaymentGateways/Routes/GatewayRoute.php

```diff
@@ -2,74 +2,90 @@
 use Give\Framework\PaymentGateways\PaymentGatewayRegister;
 
+use function Give\Framework\Http\Response\response;
+
 /**
  * @since 2.18.0
  */
 class GatewayRoute
 {
     /**
-     * @since 2.18.0
+     * This is our entry point into the Gateway Routing system.
      *
      * @return void
      * @throws PaymentGatewayException
+     * @since 2.18.0
+     * @unreleased - validate secureRouteMethods
+     *
      */
     public function __invoke()
     {
         if ($this->isValidListener()) {
             /** @var PaymentGatewayRegister $paymentGatewaysRegister */
             $paymentGatewaysRegister = give(PaymentGatewayRegister::class);
+
+            // get all registered gateways
             $paymentGateways = $paymentGatewaysRegister->getPaymentGateways();
+
+            // get all registered gateway ids
             $gatewayIds = array_keys($paymentGateways);
 
+            // make sure required params are valid
             if (!$this->isValidRequest($gatewayIds)) {
                 throw new PaymentGatewayException('This route is not valid.');
             }
 
+            // create DTO from GET request
             $data = GatewayRouteData::fromRequest($_GET);
 
-            /** @var PaymentGateway $gateway */
+            /**
+             * Get the PaymentGateway instance
+             *
+             * @var PaymentGateway $gateway
+             */
             $gateway = give($paymentGateways[$data->gatewayId]);
 
-            $allowedGatewayMethods = $gateway->routeMethods;
-
-            if (is_a($gateway, OffsiteGatewayInterface::class)) {
-                $allowedGatewayMethods = array_merge(
-                    $allowedGatewayMethods,
-                    OffsiteGatewayInterface::defaultRouteMethods
-                );
-            }
+            // get all public and secure gateway methods
+            $allGatewayMethods = array_merge($gateway->routeMethods, $gateway->secureRouteMethods);
 
+            // Make sure the method being called is defined in the gateway.
             if (
-                !in_array($data->gatewayMethod, $allowedGatewayMethods, true) ||
-                !method_exists($gateway, $data->gatewayMethod)
+                !method_exists($gateway, $data->gatewayMethod) ||
+                !in_array($data->gatewayMethod, $allGatewayMethods, true)
             ) {
                 throw new PaymentGatewayException('The gateway method does not exist.');
             }
 
-            $gateway->handleGatewayRouteMethod($data->donationId, $data->gatewayMethod);
+            // If method is in secureRouteMethods then, validate signature
+            if (in_array($data->gatewayMethod, $gateway->secureRouteMethods, true)) {
+                $this->validateSignature($data->routeSignature, $data);
+            }
+
+            // Navigate to our payment gateway api to handle calling the gateway's method
+            $gateway->handleGatewayRouteMethod($data->gatewayMethod, $data->queryParams);
```

Comment:
Yeah, why not just handle the logic here and pull it out of the gateway? This also secures protected and private gateway methods.